## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :x: [more...](#51bcf66d-619b-4105-952d-66812c0158d7) |
<a name="51bcf66d-619b-4105-952d-66812c0158d7"></a>
### 16842-cbe3e1c/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.6:30637
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 33af0a11c712:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 10 seconds
(5/14) Running awscli tests ... done in 1 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-go tests ... FAILED in 50 seconds
panic: read tcp 172.17.0.4:40738->15.15.15.6:30637: read: connection reset by peer

goroutine 1 [running]:
main.crcMatchesName({0x964160, 0xc0007fe600}, {0x839385?, 0x1e?})
	/mint/run/core/minio-go/main.go:332 +0x14e
main.testPutObjectWithMetadata()
	/mint/run/core/minio-go/main.go:2279 +0xd85
main.main()
	/mint/run/core/minio-go/main.go:12325 +0x30a
(8/14) Running minio-java tests ... done in 37 seconds
(9/14) Running minio-js tests ... done in 52 seconds
(10/14) Running minio-py tests ... done in 2 minutes and 18 seconds
(11/14) Running s3cmd tests ... done in 17 seconds
(12/14) Running s3select tests ... done in 5 seconds
(13/14) Running versioning tests ... done in 3 minutes and 33 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/internal/logger/target/http/http.go
+++ b/internal/logger/target/http/http.go
@@ -23,6 +23,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"math"
 	"net/http"
 	"net/url"
 	"strings"
@@ -222,38 +223,56 @@ func (h *Target) logEntry(entry interface{}) {
 		return
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), webhookCallTimeout)
-	defer cancel()
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost,
-		h.config.Endpoint, bytes.NewReader(logJSON))
-	if err != nil {
-		h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
-		atomic.AddInt64(&h.failedMessages, 1)
-		return
-	}
-	req.Header.Set(xhttp.ContentType, "application/json")
-	req.Header.Set(xhttp.MinIOVersion, xhttp.GlobalMinIOVersion)
-	req.Header.Set(xhttp.MinioDeploymentID, xhttp.GlobalDeploymentID)
+	tries := 0
+	for {
+		if tries > 0 {
+			if tries >= 10 || atomic.LoadInt32(&h.status) == statusClosed {
+				// Don't retry when closing...
+				return
+			}
+			// sleep = (tries+2) ^ 2 milliseconds.
+			sleep := time.Duration(math.Pow(float64(tries+2), 2)) * time.Millisecond
+			if sleep > time.Second {
+				sleep = time.Second
+			}
+			time.Sleep(sleep)
+		}
+		tries++
+		ctx, cancel := context.WithTimeout(context.Background(), webhookCallTimeout)
+		defer cancel()
+		req, err := http.NewRequestWithContext(ctx, http.MethodPost,
+			h.config.Endpoint, bytes.NewReader(logJSON))
+		if err != nil {
+			h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
+			atomic.AddInt64(&h.failedMessages, 1)
+			continue
+		}
+		req.Header.Set(xhttp.ContentType, "application/json")
+		req.Header.Set(xhttp.MinIOVersion, xhttp.GlobalMinIOVersion)
+		req.Header.Set(xhttp.MinioDeploymentID, xhttp.GlobalDeploymentID)
 
-	// Set user-agent to indicate MinIO release
-	// version to the configured log endpoint
-	req.Header.Set("User-Agent", h.config.UserAgent)
+		// Set user-agent to indicate MinIO release
+		// version to the configured log endpoint
+		req.Header.Set("User-Agent", h.config.UserAgent)
 
-	if h.config.AuthToken != "" {
-		req.Header.Set("Authorization", h.config.AuthToken)
-	}
+		if h.config.AuthToken != "" {
+			req.Header.Set("Authorization", h.config.AuthToken)
+		}
 
-	resp, err := h.client.Do(req)
-	if err != nil {
-		atomic.AddInt64(&h.failedMessages, 1)
-		h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
-		return
-	}
+		resp, err := h.client.Do(req)
+		if err != nil {
+			atomic.AddInt64(&h.failedMessages, 1)
+			h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
+			continue
+		}
 
-	// Drain any response.
-	xhttp.DrainBody(resp.Body)
+		// Drain any response.
+		xhttp.DrainBody(resp.Body)
 
-	if !acceptedResponseStatusCode(resp.StatusCode) {
+		if acceptedResponseStatusCode(resp.StatusCode) {
+			return
+		}
+		// Log failure, retry
 		atomic.AddInt64(&h.failedMessages, 1)
 		switch resp.StatusCode {
 		case http.StatusForbidden:

